incuna_auth/middleware/login_required.py

```diff
@@ -38,40 +37,25 @@ class LoginRequiredMiddleware:
     Will default to protecting everything if LOGIN_PROTECTED_URLS is not in
     settings.
 
-
     Original code from:
     http://onecreativeblog.com/post/59051248/django-login-required-middleware
 
     This version has been modified to allow us to define areas of the site to
     password protect instead of protecting everything under /.
-    '''
-    def process_request(self, request):
-        assert hasattr(request, 'user'), (
-            "The 'LoginRequiredMiddleware' "
-            + "requires authentication middleware to be installed. Edit your "
-            + "'MIDDLEWARE_CLASSES' setting to insert "
-            + "'django.contrib.auth.middlware.AuthenticationMiddleware'. "
-            + "If that doesn't work, ensure your 'TEMPLATE_CONTEXT_PROCESSORS' "
-            + "setting includes 'django.core.context_processors.auth'."
-        )
-
-        # Jump over this middleware if not a protected url.
-        path = request.path_info.lstrip('/')
-        path_is_exempt = any(m.match(path) for m in EXEMPT_URLS)
-        path_is_protected = any(m.match(path) for m in PROTECTED_URLS)
-        if path_is_exempt or not path_is_protected:
-            return
+    """
+    login_exempt_urls = [settings.LOGIN_URL, settings.LOGOUT_URL]
+    login_exempt_urls += getattr(settings, 'LOGIN_EXEMPT_URLS', [])
+    login_protected_urls = getattr(settings, 'LOGIN_PROTECTED_URLS', [r'^'])
 
-        # Jump over this middleware if user logged in.
-        if request.user.is_authenticated():
-            return
+    EXEMPT_URLS = compile_urls(login_exempt_urls)
+    PROTECTED_URLS = compile_urls(login_protected_urls)
 
-        # Raise a 403 for POST/DELETE etc.
-        if request.method != 'GET':
-            return HttpResponseForbidden()
+    def __init__(self, check=True):
```

Comment:
I could add the requisite settings to the `settings.configure` section in `run.py`, although I'd be a bit wary of breaking things.  The FeinCMS one in particular seems like a poor idea.  That one's not even installed, nor should it be...